### PR TITLE
`join-lhs-display-name` should allow `nil` `join-or-joinable`

### LINKED ISSUE #32530: [MLv2] It should be possible to call `join-lhs-display-name` without a join or a joinable
`join-lhs-display-name` added in #32420 expects either a join or a joinable (table or card metadata) argument. But when adding a new join with a notebook editor, there's a moment before a table is selected when we have neither of them. It should be possible to call the method without them to retrieve the LHS display name

cleanshot_2023-07-20_at_14 05 39_2x

### PR DESCRIPTION
**Pull request by camsaul**
Resolves #32530

### CODE REVIEW

File: src/metabase/lib/js/metadata.cljs [hunk 1]
```diff
@@ -365,15 +365,15 @@
   (some-> metadata :segments deref (get segment-id) deref))
 
 (defn- tables [metadata database-id]
-  (for [[_id table-delay]  (some-> metadata :tables deref)
-        :let               [a-table (some-> table-delay deref)]
-        :when              (and a-table (= (:db-id a-table) database-id))]
+  (for [[_id table-delay] (some-> metadata :tables deref)
+        :let              [a-table (some-> table-delay deref)]
+        :when             (and a-table (= (:db-id a-table) database-id))]
```
Comment (camsaul) [inline src/metabase/lib/js/metadata.cljs:370]: Unrelated indentation fixes

**Review by snoe — APPROVED** [review]
